chore: fix cics dialect to reflect error in correct postion in case a…

amanPras opened the pull request:
Issue is we try to adjust the dialect diagnostics . So these locality need not be adjusted.
But we don't seems to do the same on the dialect Nodes, which looks wrong. 
Plan to add that change is a different PR. This PR is to just make sure that we don't break previous cases
https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/f79f54352c8f271d2255d495d2ae4e57539fa554/server/engine/src/main/java/org/eclipse/lsp/cobol/dialects/ibm/ImplicitDialectProcessingStage.java#L84
## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test diagnostics for below code is at right location
```cobol
       IDENTIFICATION DIVISION.
       PROGRAM-ID. ABCDEF.
       DATA DIVISION.
       WORKING-STORAGE SECTION.
       copy abc.
       PROCEDURE DIVISION.
       copy abcd.
``` 
ABC.cpy
```cobol
       01 varOne   PIC S9 VALUE +10.
       01 varTwo   PIC S9 VALUE +100.
       01 varThree PIC S9 VALUE +1000.
       01 varFour  PIC X VALUE 'NAME_ONE'.
       01 varFive  PIC X VALUE 'NAME_TWO'.
       01 varSix   PIC X VALUE 'NAME_THREE'.
``` 

ABCD.cpy
```cobol
           EXEC CICS DELETE CONTAINER(VARFOUR)
           END-EXEC.
``` 
- [x] Test RETCODE is no more a mandatory option for CICS DELETE command.
check below sysprint 
```
   000160                PROCEDURE DIVISION using dfheiblk dfhcommarea.                            127 159
   000161               *EXEC CICS DELETE CONTAINER(PROCESS-NAME)
   000162               *                 PROCESS
   000163               *                 RESP(RESP-AREA)
   000164               *                 RESP2(RESP2-AREA)
   000165               *END-EXEC.
   000166                    Call 'DFHEI1' using by content x'3412200027201000000000000000         EXT
   000167               -    '00f0f0f0f1f9404040' by content x'0000' by content x'0000' by
1PP 5655-EC6 IBM Enterprise COBOL for z/OS  6.3.0 P240628       DFH0SAL2  Date 11/27/2024  Time 07:06:57   Page     6
   LineID  PL SL  ----+-*A-1-B--+----2----+----3----+----4----+----5----+----6----+----7-|--+----8 Map and Cross Reference
0  000168                     reference PROCESS-NAME end-call                                      10
   000169                     Move eibresp to RESP-AREA                                            156 7
   000170                     Move eibresp2 to RESP2-AREA.                                         157 8
1PP 5655-EC6 IBM Enterprise COBOL for z/OS  6.3.0 P240628       DFH0SAL2  Date 11/27/2024  Time 07:06:57   Page     7
``` 

## Checklist:
- [x] Each of my commits contains one meaningful change
- [x] I have performed rebase of my branch on top of the development
- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have checked my code and corrected any misspellings

amanPras committed:
chore: fix cics dialect to reflect error in correct postion in case a copybook is used

Signed-off-by: ap891843 <aman.prashant@broadcom.com>

amanPras commented on server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/VisitorUtility.java:
```diff
@@ -29,27 +28,25 @@
   public Locality constructLocality(ParserRuleContext ctx, DialectProcessingContext context) {
-    Location location = context.getExtendedDocument().mapLocation(constructRange(ctx));
-    return Locality.builder().uri(location.getUri()).range(location.getRange()).build();
+    return Locality.builder().uri(context.getExtendedDocument().getUri()).range(constructRange(ctx)).build();
```
Issue is we try to adjust the dialect diagnostics . So these locality need not be adjusted.
But we don't seems to do the same on the dialect Nodes, which looks wrong. 
https://github.com/eclipse-che4z/che-che4z-lsp-for-cobol/blob/f79f54352c8f271d2255d495d2ae4e57539fa554/server/engine/src/main/java/org/eclipse/lsp/cobol/dialects/ibm/ImplicitDialectProcessingStage.java#L84

amanPras committed:
chore: remove mandatory option check of RETCODE for CICS DELETE command

CICS DELETE command without RETCODE seems to compile well. Look below sysprint
   000160                PROCEDURE DIVISION using dfheiblk dfhcommarea.                            127 159
   000161               *EXEC CICS DELETE CONTAINER(PROCESS-NAME)
   000162               *                 PROCESS
   000163               *                 RESP(RESP-AREA)
   000164               *                 RESP2(RESP2-AREA)
   000165               *END-EXEC.
   000166                    Call 'DFHEI1' using by content x'3412200027201000000000000000         EXT
   000167               -    '00f0f0f0f1f9404040' by content x'0000' by content x'0000' by
1PP 5655-EC6 IBM Enterprise COBOL for z/OS  6.3.0 P240628       DFH0SAL2  Date 11/27/2024  Time 07:06:57   Page     6
   LineID  PL SL  ----+-*A-1-B--+----2----+----3----+----4----+----5----+----6----+----7-|--+----8 Map and Cross Reference
0  000168                     reference PROCESS-NAME end-call                                      10
   000169                     Move eibresp to RESP-AREA                                            156 7
   000170                     Move eibresp2 to RESP2-AREA.                                         157 8
1PP 5655-EC6 IBM Enterprise COBOL for z/OS  6.3.0 P240628       DFH0SAL2  Date 11/27/2024  Time 07:06:57   Page     7

Signed-off-by: ap891843 <aman.prashant@broadcom.com>

amanPras commented:
@ilidio-lopes Can you please check this PR. I have commented out the mandatory RETCODE option for DELETE command

slavek-kucera commented on server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/utility/CICSDeleteOptionsCheckUtility.java:
```diff
@@ -110,7 +110,7 @@ private void checkDeleteGroupTwo(CICSParser.Cics_delete_group_twoContext ctx) {
   private void checkDeleteGroupThree(CICSParser.Cics_delete_group_threeContext ctx) {
     checkHasMandatoryOptions(ctx.CONTAINER(), ctx, "CONTAINER");
-    checkHasMandatoryOptions(ctx.RETCODE(), ctx, "RETCODE");
+//    checkHasMandatoryOptions(ctx.RETCODE(), ctx, "RETCODE");
```
I think it should be removed from the grammar as well.

slavek-kucera commented on server/engine/src/main/java/org/eclipse/lsp/cobol/implicitDialects/cics/CICSVisitor.java:
```diff
@@ -292,7 +298,7 @@ private void addReplacementContext(ParserRuleContext ctx) {
                 context
                     .getExtendedDocument()
                     .replace(
```
Can this also `fillArea`?

amanPras committed:
chore: remove RETCODE from grammar as suggested in PR

Using RETCODE gives below error by CICS translator
00018             PROCEDURE DIVISION.
 00019                 EXEC CICS DELETE CONTAINER(PROCESS-NAME)
 00020                                  PROCESS
 00021                                  RETCODE(RESP-AREA)
 00022                 END-EXEC.
1CICS 7.2.0 COMMAND LANGUAGE TRANSLATOR                                            TIME 07.48 DATE 27 NOV 24   PAGE 2
0TRANSLATOR DIAGNOSTIC MESSAGES OF SEVERITY W AND ABOVE.
0MESSAGE NUMBER  SEVERITY  LINE NUMBER     MESSAGE
 DFH7053I           C        00019         OPTION 'RETCODE' IS NOT VALID AND IS IGNORED.
0END OF TRANSLATOR DIAGNOSTIC MESSAGES.

Signed-off-by: ap891843 <aman.prashant@broadcom.com>

slavek-kucera commented on server/engine/src/test/java/org/eclipse/lsp/cobol/usecases/TestExecCicsContainerStatementsArgumentsOrder.java:
```diff
@@ -74,7 +74,6 @@ class TestExecCicsContainerStatementsArgumentsOrder {
           + "           END-EXEC\n"
           + "           exec cics delete container({$xml-cont})\n"
               + "                     channel({$outbound-channel})\n"
-              + "                     retcode({$RESULT-LENGTH})\n"
```
This is allowed in the "EXCI DELETE CONTAINER" version of the command.
